src/flow.c

```diff
@@ -344,6 +345,30 @@ static inline void FlowUpdateTtlTC(Flow *f, uint8_t ttl)
     f->max_ttl_toclient = MAX(f->max_ttl_toclient, ttl);
 }
 
+static inline void FlowUpdateFlowRate(
+        ThreadVars *tv, DecodeThreadVars *dtv, Flow *f, const Packet *p, int dir)
+{
+    if (FlowRateStorageEnabled()) {
+        /* No need to update the struct if flow is already marked as elephant flow */
+        if (f->flags & FLOW_IS_ELEPHANT)
+            return;
+        FlowRateStore *frs = FlowGetStorageById(f, FlowRateGetStorageID());
+        if (frs != NULL) {
+            FlowRateStoreUpdate(frs, p->ts, GET_PKT_LEN(p), dir);
+            bool fr_exceeds = FlowRateIsExceeding(frs, dir);
+            if (fr_exceeds) {
+                SCLogDebug("Flow rate for flow %p exceeds the configured values, marking it as an "
+                           "elephant flow",
+                        f);
+                f->flags |= FLOW_IS_ELEPHANT;
```

Comment:
when we set this flag, we'll no longer use the flow storage, right? Might as well free it?